src/client_side_reply.cc

```diff
@@ -1484,17 +1485,18 @@ clientReplyContext::identifyStoreObject()
 {
     HttpRequest *r = http->request;
 
-    // client sent CC:no-cache or some other condition has been
-    // encountered which prevents delivering a public/cached object.
-    // XXX: The above text does not match the condition below. It might describe
-    // the opposite condition, but the condition itself should be adjusted
-    // (e.g., to honor flags.noCache in cache manager requests).
-    if (!r->flags.noCache || r->flags.internal) {
+    if (r->flags.noCache && !r->flags.internal) {
+        identifyFoundObject(nullptr, "no-cache"); // skip Store lookup for  "external" no-cache requests
+    } else if (ForSomeCacheManager(r->url.path())) { // cache manager requests ought to have flags.noCache
+        identifyFoundObject(nullptr, "no-cache"); // skip Store lookup for cache manager requests
+    } else if (r->flags.internal) {
+        // All internal requests, other from CacheManager (such as netdb or store_digest) are cacheable.
+        // Ignore "no-cache" for such requests.
+        const auto e = storeGetPublicByRequest(r);
+        identifyFoundObject(e, storeLookupString(bool(e)));
+    } else  { // "external" requests without no-cache
         const auto e = storeGetPublicByRequest(r);
```

Comment:
Please try to refactor the whole method to follow this sketch:

```C++
if (internal) {
    if (ForSomeCacheManager(r->url.path()))
        return identifyFoundObject(nullptr, "mgr"); // skip Store lookup for cache manager requests

    // all other internal requests (e.g., icons and netdb) can only be successfully satisfied from our cache
    return lookupInCache(r, "Store-dependent internal request");
}

if (flags.noCache)
    return identifyFoundObject(nullptr, "no-cache"); // skip Store lookup for no-cache and similar requests

lookupInCache(r, "ordinary external request");
```

The second lookupInCache() parameter will only be used for debugging:

```C++
void clientReplyContext::lookupInCache(... description)
{
    debugs(88, 5, description);
    const auto e = storeGetPublicByRequest(r);
    identifyFoundObject(e, storeLookupString(bool(e)));
}
```

Reply:
Done.